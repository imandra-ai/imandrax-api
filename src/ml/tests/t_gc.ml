module C = Imandrax_api_client

let pf = Printf.printf

let () =
  Printexc.register_printer (function
    | C.RPC.Error.E err -> Some (C.RPC.Error.show err)
    | _ -> None)

let client = C.connect_tcp_exn @@ C.addr_inet_local ()

let () = pf "getting GC info…\n%!"

let _gc_stats = C.Fut.wait_block_exn @@ C.System.gc_stats client

(* let () = Format.printf "GC info: %a\n%!" C.API.pp_gc_stats _gc_stats *)

let () = pf "got GC info…\n%!"

let () = C.disconnect client

let () = pf "disconnected\n%!"
