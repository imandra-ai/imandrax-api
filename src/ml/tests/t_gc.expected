getting GC info…
got GC info…
disconnected
